src/main/java/seedu/duke/command/AddLessonCommand.java

```diff
@@ -17,7 +21,17 @@ public AddLessonCommand(String title, String dayOfTheWeek, String startTime, Str
     }
 
     @Override
-    public void execute(Ui ui, TaskList tasklist, LessonList lessonList) {
-        lessonList.addLesson(new Lesson(title, dayOfTheWeek, startTime, endTime));
+    public void execute(Ui ui, TaskList taskList, LessonList lessonList) throws DukeException {
+        Lesson newLesson = new Lesson(title, dayOfTheWeek, startTime, endTime);
+        lessonList.addLesson(newLesson);
+        ui.printLessonAdded(newLesson, lessonList.getSize());
+        Storage storage = new Storage();
+        try {
+            StringBuilder dataToWrite = new StringBuilder();
+            dataToWrite.append(taskList.serialize()).append(lessonList.serialize());
+            storage.saveData(dataToWrite.toString());
+        } catch (IOException e) {
+            throw new DukeException(e.toString());
+        }
```

Comment:
Bad SLAP. Also, we have initialized `storage` inside Duke constructor.

https://github.com/AY2122S1-CS2113T-W11-3/tp/blob/b62548f5c294feb04be37e41e224306e7338fe0d/src/main/java/seedu/duke/Duke.java#L21

The `execute` method can take an instance of `Storage` as the input parameter.

Reply:
Hmm.. to save data for every command, since we have both `taskList` and `lessonList`, I think it's a bit tricky. But we can probably do something like this.

```java
storage.saveData(taskList.serialize(), lessonList.serialize());

// inside the Storage class
public void saveData(String taskData, string lessonData) throws IOException {
    // combine the two strings
    // save to a file
}
```